Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -342,7 +342,20 @@ TextureCache::TCacheEntryBase* TextureCache::Load(const u32 stage)
 	//
 	// For efb copies, the entry created in CopyRenderTargetToTexture always has to be used, or else it was
 	// done in vain.
-	std::pair <TexCache::iterator, TexCache::iterator> iter_range = textures.equal_range(address);
+	std::pair <TexCache::iterator, TexCache::iterator> iter_range;
+	if (g_ActiveConfig.iSafeTextureCache_ColorSamples == 0 || 
+		(texture_size <= (u32)g_ActiveConfig.iSafeTextureCache_ColorSamples * 8 && palette_size <= (u32)g_ActiveConfig.iSafeTextureCache_ColorSamples * 8))
+	{
+		// Allow to find this texture at a different address, if the full texture was hashed
```

Comment:
This linear search is quite inefficient. I wonder how much CPU time is spent there.

Maybe it's worth dual-indexing here and maintaining a separate hashmap?